internal/fields/mappings.go

```diff
@@ -369,51 +300,61 @@ func isNumberTypeField(previewType, actualType string) bool {
 	return false
 }
 
-func (v *MappingValidator) validateMappingInECSSchema(currentPath string, definition map[string]any) error {
+func (v *MappingValidator) validateMappingInECSSchema(currentPath string, definition map[string]any) multierror.Error {
 	found := FindElementDefinition(currentPath, v.Schema)
 	if found == nil {
-		return fmt.Errorf("missing definition for path")
+		return multierror.Error{fmt.Errorf("missing definition for path")}
 	}
 
 	if found.External != "ecs" {
-		return fmt.Errorf("missing definition for path (not in ECS)")
+		return multierror.Error{fmt.Errorf("missing definition for path (not in ECS)")}
 	}
 
-	actualType := mappingParameter("type", definition)
-	if found.Type == actualType {
-		return nil
+	errs := compareFieldDefinitionWithECS(currentPath, found, definition)
+	if len(errs) > 0 {
+		return errs
 	}
 
-	// exceptions related to numbers
-	if isNumberTypeField(found.Type, actualType) {
-		logger.Debugf("Allowed number fields with different types (ECS %s - actual %s)", string(found.Type), string(actualType))
-		return nil
+	// Currently, validation of multi-fields with ECS is skipped
+	// Any multi-field found in the actual mapping must come from a definition from the preview or
+	// a dynamic template, therefore there is a definition for it.
+	// Example of this validation can be found at:
+	// https://github.com/elastic/elastic-package/pull/2285/commits/51656120
+	return nil
```

Comment:
@jsoriano do you think if I'm missing something in this note about why this validation is skipped/ignored ?

Reply:
Looks good, thanks.